### PR TITLE
feat: implementing buy function for UniV3 MultiQuoter [LIT-763]

### PR DESCRIPTION
**Pull request by SavDont**
# Description

Implementing quoteExactMultiOutput for the UniswapV3 MultiQuoter. Appropriate foundry test cases have been added.

# Testing & Simbot Run

# Checklist

-   [ ] Update 0x API documentation (gitbook) if needed (e.g. public facing API change).
-   [ ] Add tests to cover changes as needed.
-   [ ] All dependent changes (e.g. RFQ server, Gas Price Oracle) have been deployed (if any).

### CODE REVIEW

File: contracts/src/test/UniswapV3MultiQuoter.t.sol [hunk 1]
```diff
@@ -10,7 +10,7 @@ import "../UniswapV3Common.sol";
 contract TestUniswapV3Sampler is Test, UniswapV3Common {
```
Comment (kliao) [inline contracts/src/test/UniswapV3MultiQuoter.t.sol:10]: nit: Should we put the test/ folder one level up to be the same as src/ ?

File: contracts/src/test/UniswapV3MultiQuoter.t.sol [hunk 2]
```diff
@@ -149,4 +163,26 @@ contract TestUniswapV3Sampler is Test, UniswapV3Common {
...
+        for (uint256 i = 0; i < amountsOut.length; ++i) {
+            try uniQuoter.quoteExactOutput(path, amountsOut[i]) returns (
+                uint256 uniQuoterAmountIn,
+                uint160[] memory /* sqrtPriceX96AfterList */,
+                uint32[] memory /* initializedTicksCrossedList */,
+                uint256 /* gasEstimate */
+            ) {
+                assertTrue(multiQuoterAmountsIn[i] < uniQuoterAmountIn + ERROR_THRESHOLD);
+                assertTrue(multiQuoterAmountsIn[i] > uniQuoterAmountIn - ERROR_THRESHOLD);
+            } catch {}
```
Comment (kliao) [inline contracts/src/test/UniswapV3MultiQuoter.t.sol:184]: nit: maybe we can add a console.log statement to show the two quoter outputs if the assert statement fails?

**Review by kliao — APPROVED** [review]

File: contracts/test/UniswapV3MultiQuoter.t.sol [hunk 3]
```diff
@@ -4,13 +4,13 @@ pragma experimental ABIEncoderV2;
 import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
 
 import "forge-std/Test.sol";
-import "../UniswapV3MultiQuoter.sol";
-import "../UniswapV3Common.sol";
+import "../src/UniswapV3MultiQuoter.sol";
+import "../src//UniswapV3Common.sol";
```
Comment (kliao) [inline contracts/test/UniswapV3MultiQuoter.t.sol:8]: is this a typo?

**Review by kliao — APPROVED** [review]

File: contracts/test/UniswapV3MultiQuoter.t.sol (hunk 3 above)
Comment (SavDont) [reply contracts/test/UniswapV3MultiQuoter.t.sol:8]: good catch 🙏🏾

**Pull request merged**